### ISSUE TITLE
Replace ownCloud with something more stable

### ISSUE THREAD

**Issue by JoshData** [mail-in-a-box/mailinabox#514]
We've had [dozens of issues](https://github.com/mail-in-a-box/mailinabox/search?q=owncloud&type=Issues&utf8=%E2%9C%93) with ownCloud. Most recently, editing events is totally broken (https://github.com/owncloud/calendar/issues/892) -- fortunately I noticed before making a new release of Mail-in-a-Box. Prior to that, ownCloud [broke their release distributions](https://github.com/mail-in-a-box/mailinabox/issues/453) and put one version under the name of a different version, and clobbered their own hash signature to hide the problem.

They're making me waste a lot of time trying to pin down their problems.

So I'm now going to be looking into replacing ownCloud with [SabreDAV](http://sabre.io/), which is ownCloud's DAV component, plus some new calendar/contacts front-end.

In this scenario, the Files app will be dropped with no replacement (it's not something I ever intended to support).

Suggestions welcome.

**Comment by mbardelmeijer** [comment]
:+1:

**Comment by anoma** [comment]
I have to agree. From my use case all I have ever wanted is reliable contacts, calendar and notes that can be shared between users. I think this a natural match to mailinabox. Unfortunately the shared part seems very hard to achieve outside of OwnCloud and as we know it spends as much time broken as working.

When experimenting for alternatives I particularly liked http://baikal-server.com/ 

Interestingly on looking again it appears at least some work on shared contacts and calendars is in the works (https://github.com/netgusto/Baikal/issues/303)

**Comment by dhpiggott** [comment]
I'd be happy with a move away from ownCloud to SabreDAV.

I wouldn't be so sure about Baikal. Before I switched to MIAB I used Baikal to meet my *DAV requirements, but I found on a number of occasions that my contacts had been corrupted. Unfortunately my memory of the corruption issues I had isn't great, so I can't say with certainty that it was Baikal (I may have concluded it was an issue with CardDAV Sync which I was using at the time). It might even have only happened with [Radicale](http://radicale.org/) which I used for some time before Baikal (maybe that was why I switched!). What I do know is that I've not had any corruption issues with MIAB + DAVdroid.

**Comment by PortableTech** [comment]
I certainly have my own frustrations with the OwnCloud calendar app where a deleted calendar can not be properly recreated.  They certainly could do a better job on things.  

That being said, my group has become rather hooked on the file sharing features so I would certainly hate so see that option go.  I am sure I could find some other way to do it if I have to make a separate server for that purpose, but it sure was nice having it all in one place and using a common authentication scheme.

**Comment by RobWW** [comment]
One feature of Owncloud I really find useful is the calendar sharing.  I looked at SabreDAV, but found it unclear if this capability is implemented.  Hopefully this feature could be retained in any Owncloud replacement.

**Comment by anoma** [comment]
I am struggling to find information about shared contacts, calendar and notes using SabreDAV. Obviously it is possible since OC do it but none of the other Card/Caldav projects seems to support it which is odd since they essentially all use SabreDAV as a back end. Its is doubly odd since you can pick any of these projects and find a request for it.

To me this and dropping filesharing are likely the only two viable push backs we would see from dropping OwnCloud.

**Comment by DeviantEng** [comment]
I support this transition.

I run OwnCloud on my own network for file backup, and use the OwnCloud with MiaB only for contacts and calendar.  I run MiaB on a small-ish VPS, and have limited room for file storage, so it was never going to be a file storage box for me.  

+1 to ditch Owncloud and move to SabreDAV!

**Comment by JoshData** [comment]
And the latest from ownCloud: https://github.com/owncloud/core/issues/18224

**Comment by nstanke** [comment]
@JoshData Couldn't we run `occ upgrade`[1] twice. (As a workaround to release v0.13)

[1]https://github.com/mail-in-a-box/mailinabox/blob/master/setup/owncloud.sh#L73

**Comment by JoshData** [comment]
Let's take that conversation to https://discourse.mailinabox.email/t/v13-upgrade-issue-with-calendar/757/2.

**Comment by jonathannz** [comment]
@JoshData I feel the same way. Found you're post whilst looking for alternatives. For the past year or so I've had to spend hours (literally) messing around with getting Contacts and/or Calendar to work each time I update OC. Sometimes Con and Cal are not able to be updated and/or enabled, and I go through a long winded process to uninstall and install them, along with various suggested hacks. I keep hoping the situation will improve. Added to that is the fact I've been having issues with Contacts not updating... that's been ongoing for many months. Lost count.
Anyway... enough of a rant. I'll be looking into SabreDAV. Sharing of Calendars is a must for me though. Can anyone in this discussion confirm if it supports that?
Thanks...
Jonathan

**Comment by hnk** [comment]
re-post from the discourse forum:

>While I like owncloud for the file-sharing part, I have always been, ... sceptical, about their QA.
>
>Here is a good list of calDAV/cardDAV servers, just for reference
>https://github.com/Kickball/awesome-selfhosted#calendar-and-address-books4
>
>I know you never wanted the whole file sharing part in miab, but I think it is a good fit for an email server, as it does allow the management and sending of larger attachments via email (as links) in a controlled environment. Many people need to send files around regularly, and offering that is a great thing IMO.
>
>Maybe someone can give input to something like pydio2. I have read good things about it.

**Comment by PortableTech** [comment]
I could certainly live without owncloud if we had some other, ideally integrated, file storage method.  As was mentioned earlier, the ability to move large files into a share location, or store easily accessible common repositories makes things so much simpler, and honestly keeps mailboxes smaller and more manageable.  Far to often without such a solution, I see people take that large powerpoint document, or other such item and email it to 50 people, now it takes 51x the space (counting the sender) of the file size in the mail stores.  That just does not seem idea.

Add to that that many commercial email services out there set their attachment limit to 10MB, or sometimes, even 5MB, email can become quite limiting by itself.

I did look at Pydio as suggested earlier and it looks nice, honestly cleaner than owncloud.  I think it is well worth consideration.

In the end it would be great if we could get someway for RoundCube to be able to sync with the contacts list for ActivSync.  Right now I just export and import it from RoundCube to owncloud from time to time, but automating that would make end users much happier.

**Comment by anoma** [comment]
So as I interpret the discussion to date, any replacement  for OwnCloud,

Must provide:
* calendar
* address book
* multi user file sharing

lack of this following features will count as a regression for many users (me included):
* multi user shared calendar
* multi user shared address book

This to me sounds like we may be looking at using multiple projects and joining them with central authentication. Whilst this all may seem pretty obvious it is nice to see that no one is making a case for any other unofficial OwnCloud addons.

Also technically the only file sharing requirement so far could be handled with maildir de-duplication.

**Comment by PortableTech** [comment]
I received the following recommendation from a friend of mine.  I have not used it myself, but it does seem to be somewhat developed on GitHub.

http://baikal-server.com/

Baikal seems to focus on doing Address Books and Calendars well, and nothing else.

**Comment by anoma** [comment]
There is some reference to shared Address Books and Calendars on the development branch. See third post of this thread. However for the best part of a year there are no code changes linked to that ticket or any movement on the whole  branch in 6 months.

Apart from that it is an excellent product.

**Comment by hnk** [comment]
Baikal looks nice, but from what I understand there is no way to define an external authentication system. Also looking [at the Roadmap](https://github.com/netgusto/Baikal#roadmap) I can see that 'import' and 'export' features are still only planned. This would be a blocker for migration.

The only project that I could find that handles imap authentication is [radicale](http://radicale.org/). It does calDAV and cardDAV.
There is an option for rights management, but it is file based. but it should be possible to get this connected to a web interface somehow to manage calander sharing.

sabre/dav is also a nice option and I found a comment that proposes a [proper way to authenticate via imap](https://github.com/netgusto/Baikal/issues/52#issuecomment-111258570) on the baikal github issues page.
Combined with an online calendar browser like [AgenDAV](http://agendav.org/) I think it is possible to get a good working solution.
But the sabre/dav documentation says that [calendar sharing is hard to implement](http://sabre.io/dav/caldav-sharing/). That is a minus.

Combining something like sabre/dav with agenDAV and a cardDAV plugin for roundcube [like this one](http://plugins.roundcube.net/packages/roundcube/carddav) would provide a good solution with tools that are built to do one thing, but do it right.

But I think there must be a lot more discussion about the tools in question and more inspection about the viability of a switch to a new tool. Questions like
- Can the new tools be installed automatically via our available toolset?
- Can the new tools be configured automatically via our available toolset?
- Can we migrate existing installations to the new tools, or
- Do we have to create a new, backwards incompatible, version of miab and if so
- Does that old version receive bug-fixes and security patches as well?

come to mind. This is something that needs to be well evaluated by @JoshData for each available tool option before making a decision.

I am more than happy to test tools and see how well they might integrate with MIAB in my spare time.
If you ( @JoshData ) want to define a set way of going about this task, let me know.

**Comment by JoshData** [comment]
The only absolute requirements are an address book and a sane upgrade path. Calendar, calendar sharing, and file sharing are nice-to-haves, but let's not get carried away.

So far I'm leaning toward Sabredav because it means a clean separation of the backend storage and UI, which means we can swap in any UI as we like, and since it's the backend within ownCloud we know it's mature (and as far as I can tell, not related to any of ownCloud's issues).

**Comment by DeviantEng** [comment]
I agree with @JoshData.  Contacts/address book is a must, but the rest are definitely nice-to-haves.

**Comment by anoma** [comment]
Its not as easy to say "nice-to-haves" when you are taking features away users already have and need.

But I have to agree, more than anything OwnCloud needs to go.

**Comment by nstanke** [comment]
@JoshData Did you start working on a Sabre/Dav integration?

**Comment by JoshData** [comment]
No. I don't know when I'll have time for it either yet.

**Comment by nstanke** [comment]
@JoshData Thanks for letting me know. I'll start working on a POC then.

**Comment by glennbb** [comment]
How tied in with the rest of the system is owncloud? Would it be easy to alter the install to prompt whether to install or bypass owncloud?  This could be temporary fix until Sabre/Dav is working.

**Comment by anoma** [comment]
For what it is worth I just upgraded my local dedicated OC setup and again it broke calendars and contacts. I have fixed them yet again but it adds weight to the argument that the bit we actually use in MIAB is the least reliable part of OC.

Also it would seem that you cant share contacts anymore either (essentially breaking the entire contact app for me).

**Comment by bzg** [comment]
I've not read the full thread but I don't see CozyCloud (https://cozy.io) mentioned here.  Maybe too full-featured for the purpose at hand, but worth considering?

**Comment by brylie** [comment]
:+1: for Cozy.io

**Comment by frankrousseau** [comment]
:+1 for Cozy :)

**Comment by h8h** [comment]
Cozy is quite nice, but can you share you calendars? As far as I know you can invite friends for each events, but not share the whole calendar with them.

**Comment by PortableTech** [comment]
Cozy looks pretty promising.  will have to play around with it some more.

**Comment by nstanke** [comment]
-1 for Cozy. Because it's in NodeJS.

**Comment by anoma** [comment]
Attractive to look at but it seems you cant share contacts either.

**Comment by zoof** [comment]
What about DAVical?  It appears to still be under active development and I've used it for years for both calendaring and contacts.  There are two attractive web based front ends: caldavzap and caldavmate.

**Comment by nstanke** [comment]
I absolutely would vote for DAVical, seems to be by far the best calendar implementation, but I guess PostgreSQL is to clumsy for MiaB.

**Comment by anoma** [comment]
I had never even seen it before and playing with the demo and reading its specs I like what I see so far.

Will try and get time to test it properly ASAP

**Comment by zoof** [comment]
And for contact integration with roundcube, I used https://github.com/christian-putzke/Roundcube-CardDAV. There are also calendar plugins that are supposed to work with caldav that I haven't used -- none of my users are interested and I use Thunderbird/Lightning.

**Comment by rockman73** [comment]
Maybe all MIAB needs is a really awesome MAIL server and nothing else for now? Fruux is a perfectly acceptable service to use as a workaround. Sync.com for cloud storage. 

Perhaps allow the admin to (un)select ownCloud from the MIAB installation? This would be a nice option for those who just need a MIAB. :)

**Comment by aspdye** [comment]
or we just work together with the ownClouders to create someting Great ;)

**Comment by rockman73** [comment]
Sure, but right now it appears owncloud is unreliable. Why used a hacked-up Card/CalDAV implementation when Fruux (the folks who maintain SabreDAV) works nearly perfectly? Isn't the point of MIAB is an easily to stand up and maintain email server? "Take back control of your email with this easy-to-deploy mail server in a box." owncloud just degrades the experience in this context.

**Comment by brylie** [comment]
The cloud storage and collaboration aspects are really important as well. In essence email is only one part of the picture. We need a unified system, so that we can liberate ourselves from GMail, Google Drive, and Google Calendar.

On 12 November 2015 03:08:54 EET, rockman73 <notifications@github.com> wrote:
>Sure, but right now it appears owncloud is unreliable. Why used a
>hacked-up Card/CalDAV implementation when Fruux (the folks who maintain
>SabreDAV) works nearly perfectly? Isn't the point of MIAB is an easily
>to stand up and maintain email server? "Take back control of your email
>with this easy-to-deploy mail server in a box." owncloud just degrades
>the experience in this context.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/mail-in-a-box/mailinabox/issues/514#issuecomment-155965351

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by dhpiggott** [comment]
@rockman73 I don't doubt that Fruux build a high quality product and provide a good service. But to me CalDAV & CardDAV are core to what it is to provide email services. As I perceive one of MIAB's goals to be to enable [redecentralization](http://redecentralize.org/), it should continue providing integrated CalDAV & CardDAV support so that our boxes remain independent appliances.

At risk of going off at a tangent, but to illustrate my own feeling about how key CalDAV & CardDAV are, it continues to surprise me that Thunderbird does not integrate CalDAV & CardDAV support and that it has to be added by plugins. Which is partly why I use Evolution instead.

I'd sooner see ownCloud left in place than removed entirely -- bugs or not. I've not actually been affected by the ownCloud issues, presumably because I don't use its web UI at all -- I just sync Evolution and DAVdroid with it. Which is why I'd be equally happy for it to be replaced with a separate backend and frontend (or not, because we could always add one later -- in an ideal world, there'd be a single web UI that could replace Roundcube and the ownCloud UI, providing email, contacts and calendaring all in one).

@nstanke What do you mean by "I guess PostgreSQL is to clumsy for MiaB"?

**Comment by JoshData** [comment]
> We need a unified system

What people need isn't the issue. People need a lot of things. The limiting factor is that I don't have time to be the maintainer of a project that takes up more of my time. So as I've said, I want Mail-in-a-Box to provide an address book (because you can't email people if you don't know their address). We've gotten calendars and files along for the ride so far, but if dropping those features makes my life easier, then I'll do that (but we'll find a way to offer users an upgrade path to some other service).

**Comment by PortableTech** [comment]
Well said Josh.  Mail is clearly our core concept here, and address books are core to that.  In this statement there is no doubt.  As to Calendars and File Sharing, they are nice, really nice.  They are what many of us view as a collaboration suite, and clearly is the path that the majority of providers have taken, but that does not mean it has to be completely unified.  

As to Calendar, it is not even currently integrated into roundcube, not directly anyway.  File Sharing is not either, but there are a ton of third party solutions.  If we can at minimum work out the ability for third party services, of which we have named several in this thread, to work with the existing user database, then there can always exist the option for individuals to integrate and manage independently of the MiaB control panel the other services.  

Personally I like ownCloud, but sadly like so many things that try to do everything well, they tend to do mediocre at best.  At it stands, I have been fortunate in not experiencing the majority of bugs, but I know others have.  Some of the other solutions that have been discussed focus on doing one thing well, and often that can lead to a better result. much like the core install of MiaB does it's job very well.

No matter which way this all goes in the end, your efforts continue to be greatly appreciated Josh as well as all those who help contribute to this project.

**Comment by DeviantEng** [comment]
Mail and carddav are the primary needs, IMO.  

I run MiaB on a VPS with a 30GB drive, so after web site space, and mailbox space, room for file storage is very little.

**Comment by brylie** [comment]
> The limiting factor is that I don't have time to be the maintainer of a project that takes up more of my time.

How could more developers contribute to maintaining this project? What are some ways we can encourage community stewardship?

**Comment by anoma** [comment]
Devils advocate. Could we work on DAVical as a standalone docker as both a proof of concept of DAVical itself and potentially using docker containers as modules to compliment traditional apt-get'ing required components.

My thought is that if viable this modular approach makes the effort to assist modular as well reducing the very steep curve to offer any meaningful dev help here.

**Comment by dhpiggott** [comment]
Nice idea -- using Docker for *DAV services. I've kept out of all discussions around Dockerising MIAB until now, but flipping it around like this is something I could support -- using Docker only for some aspects of MIAB, where it can be advantageous to do so. It could simplify prototyping/development of the ownCloud replacement.

I nearly commented asking why you favour DAVical and what we would use to provide CardDAV support because from the name and first glance of the website it sounded like DAVical doesn't provide it. In case anyone else nearly makes the mistake though, [this bit](http://www.davical.org/clients.php) of the website corrected my misunderstanding:
> DAViCal creates two collections automatically when a user is created. In recent versions these are called 'calendar' and 'addressbook'. Some software also makes it easy to create more calendars and addressbooks, or you can create more through DAViCal's web interface, also.

**Comment by JoshData** [comment]
@brylie I appreciate you asking that. Answering questions on the forum, helping to triage/fix issues and pull requests, and testing possible future changes (like Davical) would all be helpful.

@anoma I don't think mixing issues is going to help, and I really don't see a future for docker here anymore. (#590) (That's not a pronouncement - just a prediction.)

**Comment by zoof** [comment]
I just got DAVical working on a different VPS instance so I'm happy to help with putting together a pull request -- I will need some help/guidance though since I have no familiarity with nginx and I'm not sure what I can install and run in MiaB without clobbering something else.

**Comment by anoma** [comment]
@JoshData I see what you are saying but I dont think we are mixing issues. The problem is most of us could install Davical and develop with it but few of us could integrate it into the complex MIAB appliance. Also in doing do you pollute the appliance itself making it no longer MIAB. Net result few people are doing any dev.

However if you could have DAVical working within MIAB but isolated in a container simply by issuing a "docker run miab/davical" command many of the hurdles and problems go away.

As if to highlight my point @zoof just hit this exact scenario. This is not a coincidence it is exactly what docker is designed to do.

**Comment by JoshData** [comment]
Ok, sure, for testing, whatever works. :)

**Comment by dustinboston** [comment]
Just wanted to mention that I am using ownCloud for syncing files as well as contacts and calendars. I figure the better stats you have about usage, the easier it will be to make the decision. :-)

**Comment by aspdye** [comment]
Let's wait until the ownCloud 9.0 release. CalDAV and CardDAV will be part of the owncloud/core, and the CalDAV and CardDAV Apps "Contacts" and "Calendar" will be frontends-only. 


The "Clients":
https://github.com/owncloud/calendar-rework
https://github.com/Henni/contactsrework

The "Server":
https://github.com/owncloud/core/pull/20371
https://github.com/owncloud/core/pull/20371

We can hope, that ownCloud (especially calendar and contacts) is much more stable then ;) 
ownCloud Mail will be there with v0.4 too... and maybe it could be a "long-term" replacement for RoundCube...

**Comment by jospoortvliet** [comment]
Note that for ownCloud 9.0 we're bringing CalDAV and CardDAV (which were entirely community/volunteer projects until now) into ownCloud core so we can give them more attention. That should help with their stability, for sure. Of course, patches are always, and still, welcome. It might be less work to patch an existing, imperfect technology than build something new...

On the upgrade process, we're working on fixing that one, too: https://owncloud.org/blog/making-owncloud-upgrades-more-robust/

Of course, nothing is ever perfect and anybody who thinks they can improve is welcome to do so, within ownCloud or as something new.

**Comment by bronson** [comment]
@jospoortvliet in addition to the upgrades, owncloud release management has been awful.  Modifying released files and changing hashes makes trustworthiness a big issue.

**Comment by anoma** [comment]
@jospoortvliet in my opinion this is a very smart idea.

I could go on about how one of them break in some new interesting way after every update or that I dont understand the inconsistency why you can no longer share contact groups between users but you can share calendars.... but I suspect these are not new complaints.

I will say that in my circle everyone wants OwnCloud to just work but universally we have all stopped updating and now run it only internally because it just is SUCH a time drain keeping it working between upgrades.

I applaud anything that addresses the above.

**Comment by jospoortvliet** [comment]
@anoma yes, the calendar and contacts app have had issues, that is why we pull the core functionality in ownCloud proper. Understand that these apps were developed by students in their spare time and a company or university depending on it, well, that might not be wise.

For enterprise use, there's ownCloud.com which can help and they have deployment tools, advice and support which should make it a lot easier on sysadmins to run an ownCloud instance easily and safely.

@bronson yeah, we had our issues. As that is process and people rather than code I can't point to any proof of improvement but we're obviously working on that, too.

**Comment by bronson** [comment]
I'm just glad to hear you recognize it's an issue and are working to fix it.  I hope it goes well.

**Comment by MrFloppy** [comment]
Just my two cents: For file sharing I really like Seafile (https://www.seafile.com/en/home/). Perhaps this is worth a look.

**Comment by yodax** [comment]
I use seafile as well. Quite happy with it. Especially the remote encryption feature. I do have it on a separate server though.

**Comment by h8h** [comment]
Hey,

I wasted my last two days to create a PR for miab but I've failed ... maybe there's hope :). 

I would like to replace owncloud and roundcube with [SOGo](http://sogo.nu). 

I know there is also a closed issue #644 about not using SOGo but I think SOGo makes our work easier not harder, let me explain it:

They released version 3.x and it's very fancy and shipps a caldav / carddav server. It also has support for sieve filtering and a very nice and easy to use vacation plugin. So we don't need owncloud and roundcube any longer and can save the download tar files and move them around effort. Everything we need is shipped in deb files by the maintainers. We even don't need a ldap server.

Heres a [demo](http://demo.sogo.nu/SOGo)
username: sogo1
password: sogo1

The only cons I figure out are:
  * can't authenticate against dovecot, so it needs access to the `users.sqlite`
  * don't support sqlite3 so the users.sqlite have to be at least a mysql or postgresql database.
  * don't support SHA512-CRYPT, so we have to use something else (maybe less secure), see bug [2682](http://sogo.nu/bugs/view.php?id=2682).

Things I've done the last two days:
  * established a postgresql server with user peering, so that we don't need to hassle around with passwords. 
  * rewrote mailinabox-daemon to use postgresql database (the sql statements are quite the same, so thats not a big deal)
  * rewrote dovecot / postfix  ... i've got some troubles with postfix, maybe it was too late, but its supposed to work. There are some sql statements they need an additional parameter, but thats also no big deal.
  * setup [nginx](http://wiki.sogo.nu/nginxSettings) is quite easy.
  * setup sogo config to use the `user` table as `SOGoUserSources`. It failed because sogo does not support SHA512-CRYPT right now, see bug [2682](http://sogo.nu/bugs/view.php?id=2682) 

Maybe we can discuss if SOGo is worth to put more work into it or if its only a waste of time :). 

Furthermore if its a good idea to use a database (postgresql) and if its ok to use something else instead of SHA512-CRYPT, maybe SSHA512 (salted sha512) (see [source code](https://github.com/inverse-inc/sogo/blob/master/SoObjects/SOGo/NSData%2BCrypto.m#L239)), dont know.

Thanks!

**Comment by JoshData** [comment]
I appreciate your effort, but those are all deal breakers:

> can't authenticate against dovecot, so it needs access to the users.sqlite

Having a separate auth path will make implementing two-factor authentication later considerably harder. Ok this isn't a deal-breaker, but it is significant.

> don't support sqlite3 so the users.sqlite have to be at least a mysql or postgresql database.

I really don't want to run a full database. It makes administration considerably harder.

> don't support SHA512-CRYPT, so we have to use something else (maybe less secure), see bug 2682.

Migration to a new scheme will be extremely painful for existing users (every user will have to reauthenticate). The fact that they only support `CRYPT-MD5` also indicates they are not paying proper attention to security, as MD5 has been broken for like a decade.

**Comment by h8h** [comment]
Thank you for your fast feedback.

SOGo has the best solution for our purposes, but I fully agree with your so called `deal breakers`.  Its a real bummer that such a great tool has such issues.

**Comment by anoma** [comment]
Just some follow on info. According to https://github.com/owncloud/core/wiki/Maintenance-and-Release-Schedule OwnCloud 9 is due for release on the 8th Match 2016.

I think at this point if the better upgrading stability is being actively worked on and the adoption of the contacts and calendar functions into core materialize in v9 I see no reason to even entertain dropping OwnCloud anymore.

However, and this is aimed at @jospoortvliet, I cannot find any reference to the contacts and calendar move anywhere. **Can you confirm it is happening in version 9**

**Comment by aspdye** [comment]
I can confirm that there will be changes to **CardDAV** and **CardDAV** in **ownCloud 9.0**, which will release in 03/2016. 

CalDAV and CardDAV are already integrated in the current ownCloud Core "master"-build:
https://github.com/owncloud/core/pull/20371
https://github.com/owncloud/core/pull/20184

The Frontends (Calendar and Contacts Apps for ownCloud 9) are currently under heavy development and I think they will be finished until the release of ownCloud 9. They will replace the current Contacts and Calendar-Apps with ownCloud 9, and should bring more stability (together with the CalDAV and CardDAV Backends in Core) to the whole system.

https://github.com/owncloud/calendar-rework
https://github.com/owncloud/contactsrework

If everything works - you are right @anoma - there would be not longer any reason for dropping ownCloud. I hope so :rocket: 

But I want to make the addition, that we should wait until ownCloud 9.0.1 or 9.0.2 before we integrate it into Mail-in-a-Box. Then the most "common" bugs will be fixed, and we can call ownCloud stable for production use.

Also: We cannot upgrade our ownCloud Instance from 8.1 (currently integrated in MiaB) to 9.0. We must upgrade to 8.2 first - and then - we could upgrade to 9.0. We should plan this with our scripts. 

As @jospoortvliet already said, the upgrade process will be "better" with the coming releases, so that these problems will be gone.

**Comment by anoma** [comment]
@aspdye your git and google foo is stronger than mine, I failed to spot any of this. It is very encouraging to see the level of work going on for these core features. I agree we should wait until the first point release, I suggested on a ticket somewhere this very idea.

**Comment by aspdye** [comment]
As soon as ownCloud 9.0 will be released I am going to fork Mail-in-a-Box for bringing it with working Calendar, Contacts and Exchange ActiveSync into Mail-in-a-Box as stable and fast as possible :wink: 
I am already testing the pre-alphas with mail-in-a-box.
My own Build is working great yet - apart of the "normal" pre-alpha bugs and missing features :muscle: 
As soon as ownCloud publishes the first BETA of ownCloud 9.0 - I will create a BETA of Mail-in-a-Box with ownCloud ;) So stay tuned!

**Comment by anoma** [comment]
Excellent.

Quickly reading back then I think OC9 will cover every official requirements discussed in this ticket.

File sharing is still a grey are at best. I will assume that since it can be made to work now we will probably be in the same position in the future. Might be worth a new discussion around this for the long term.

Can you confirm if Calendar and Contacts are allowed to be shared between groups and users again. This "on again off again" OC feature is one I personally believe is a "must have" for mail in a box.

**Comment by aspdye** [comment]
CalDAV and CardDAV sharing is planned and will be integrated ;)

**Comment by dustinboston** [comment]
+1 for keeping owncloud

**Comment by PortableTech** [comment]
Agreed, given the new news, keep OwnCloud for now.

**Comment by laupkram** [comment]
I am facing the issues with regards to OwnCloud during installation of MiaB

`fatal: reference is not a tree: <then a hash code>`

**Comment by anoma** [comment]
OwnCloud 9 has been released https://owncloud.org/nine/

I suggest we stick with the policy of waiting until the first point release for inclusionbut independently of this we should check if it fulfills the features we expected.

**Comment by aspdye** [comment]
I am working on it, and will keep you informed. The work of the ownClouders done with ownCloud 9 is great - but there are some aspects which should fixed before we are going to release it as the side of Mail-in-a-Box. To make an Upgrade possible - we should Update the MiaB Instances to 8.2 - because otherwise - an direct Upgrade to ownCloud 9 is impossible :wink:

**Comment by anoma** [comment]
I have managed to do some light testing of the new contact and calendar applications on OC9.

They are not as feature complete yet as the older incarnations however they are immediately and obviously more stable. They also both support sharing contacts and calendars between users and it actually seems to work for once.

Based on the number of features and fixes that are due to drop for both in the next few weeks i suggest strongly that we stick to out policy of waiting until the first point release but so far it gets a huge thumbs up from me and we should look to updating when the time right.

**Comment by aspdye** [comment]
Nothing different to that what i have said above. For an Production release, we should and will wait until 9.0.1 or 9.0.2 :wink:

**Comment by groentebroer** [comment]
Wish I had read this earlier. Spend some time restoring ownCloud 8.x after upgrading and failing hard at ownCloud 9. Even tried to apply patches that already had been done on ownCloud 9, but there are just to many patches!

What are these guys doing? Just forgot about groups and sharing between groups? As I understood ownCloud became more professional because of the uptake it had. When I look at this now, that certainly does not seem to be the case....

These issues really gave a headache after update:

https://github.com/owncloud/core/issues/22932
https://github.com/owncloud/core/pull/23342
https://github.com/owncloud/core/pull/23318
https://github.com/owncloud/core/issues/23317
https://github.com/owncloud/core/pull/23398

How could these be "forgotten"?

**Comment by DeepDiver1975** [comment]
> What are these guys doing?

Thinking about dozened of combination and scenarios - and as we are humans as well we forget things. But it's always the same - as soon as one thing breaks everybody is attacking this single spot.

And just in case you want to start screaming "How can you forget such an central piece?" - Did you test the new webdav endpoint remote.php/dav/? Things work perfectly there - we did miss to have a closer look on the legacy endpoints. Shit happens - sorry for being human.

**Comment by anoma** [comment]
@DeepDiver1975 my read on this was more a "rage at the sky" type thing than a direct complaint. :)

... but regardless take a healthy +1 compliment from. My whole extended family thanks you for this as I have been looking for something reliable for what seems like a lifetime now.

**Comment by aspdye** [comment]
...and to quote @DeepDiver1975 : "Its OpenSource. Contribute. THX." :wink:

**Comment by F0rth** [comment]
http://www.horde.org/ modular, caldav/carddav server provided through sabredav.

**Comment by aspdye** [comment]
☝️test, develop and enjoy 😉

**Comment by dustinboston** [comment]
:clap: :clap: :clap:

**Comment by anoma** [comment]
http://www.phoronix.com/scan.php?page=news_item&px=OwnCloud-Forked-To-NextCloud

More issues in OwnCloud land, at some point we have to cut our losses this ticket is 10+ months waiting for OC fixes.

**Comment by msgerbs** [comment]
NextCloud is a fork of owncloud, I imagine switching to that is probably the simplest solution to that problem.

**Comment by anoma** [comment]
As far as I can tell the developers of the code we most care about (Contacts and Calendars) are still working with OwnCloud.

At the very least there will be 3+ months of turmoil as things settle down taking us over the 1 year mark of waiting.

Meanwhile we have real a PR for Radicale https://github.com/mail-in-a-box/mailinabox/issues/837 which I personally have no experience with but appreciate the move in a positive direction.

OwnCloud has always been a square peg for a round hole for this project with the promise of being able to hammer into place always around the _next_ corner.

**Comment by msgerbs** [comment]
Well, not quite. OwnCloud, Inc. has dissolved and presumably the remaining developers joined NextCloud. I do see your point but many people appreciate having everything including file sharing all in one place. It would seem silly for me to set up an OwnCloud server separate from MIAB and have calendars and contacts in two different places.

-------- Original message --------

**Comment by anoma** [comment]
I am not seeing what you are seeing then as the press releases and interviews all have a line such as `Frank Karlitschek is forking ownCloud to create a new open source project called Nextcloud`. It looks like a typical fork to me.

We wont know for sure until the codebase is released in July.

I have no problems using *Cloud, I never had, I am just getting annoyed with the seemingly perpetual wait for a product that is actually production ready for the core feature we need.

At what point do we simply say enough is enough?

**Comment by jospoortvliet** [comment]
Due to the positive responses we've been getting in the community, we sped up our plans and we've released Nextcloud 9. It's easy to migrate if you like. Yes, we'll put more efforts in Calendar and Contacts in the future, same with the Mail app and communication and other things.

Of course no software is without bugs so perfection will remain elusive.

**Comment by anoma** [comment]
For what it is worth I revert my position on the NextCloud fork.  My initial comments were based on the little information that was out there at that point but having taken time to read in more detail, assuming what is being written is the accurate, I fully expect NextCloud to be a viable (and potentially the only) *cloud option in the very near future

The question is do we commit to it. Certainly we cant run OC8 for too long regardless so in my humble opinion we make a decision soon based on the information we have and revisit it again in a year.

What I will say is how annoyed will be be if we move away from *Cloud only to find in 6 months time its the Grail project everyone always hoped it will be.

On a personal note, any annoyance that comes across in my writing is purely driven by the family asking "are we there yet" with the shared contacts an calendar i promised 2 years ago and not a dig at anyones code.

Long live FOSS.

**Comment by jeffmcneill** [comment]
There should be a hard deadline. Obviously this is emotionally driven, based on alternating bouts of hope and disappointment. A hard deadline, past which Owncloud has either delivered or not delivered xyz functionality. If not delivered, then a suitable time to determine an alternative solution and pursue that. The idea that Owncloud will either become more stable/stable enough and more functional/functional enough is great but likely unrealistic, as projects go through cycles of stability/instability, especially and profoundly those with such a large feature set as Owncloud.

**Comment by leakypixel** [comment]
Sandstorm.io doesn't seem to be well-known, but I feel it could be a good fit. It's main strength is that services, such as calendar, are added as "apps". Could be a crowd pleaser.

**Comment by just4t** [comment]
You may wish to pay a look at Nextcloud as a potential alternative to ownCloud, perhaps, when getting ready for Ubuntu 16.04 (Not sure if MiaB could live together with MySQL & PHP7 in the future)

- Focused on Privacy and promising additional features like External storage encryption and Secure audio & video calls: https://nextcloud.com/features/
Github: https://github.com/nextcloud/server

Edited 161009-1454 UTC: Supports SQLite, too

P.D. : link to the [MiaB discourse](https://discourse.mailinabox.email/t/nextcloud-alternative/1551?u=just4t) post.

**Comment by bronson** [comment]
@just4t, Nextcloud puts more drama into an already overly-dramatic situation.  See this thread for some recent comments.

https://github.com/mail-in-a-box/mailinabox/pull/894#issuecomment-245902916

Personally, I love the idea of decoupling MiaB from *Cloud and switching to something more stable.  (a LOT more stable...)

**Comment by anoma** [comment]
FWIW I have recently moved locally to NextCloud and have so far had nothing but out of the box success.

(Given that I have been pretty vocal in the past with the headache actually living OC as a sysadmin was this is a complete turn around for me).

But really i grow tired of this now multi year debate. We should just commit to one route and as none are perfect just live with the downsides once and for all.

**Comment by just4t** [comment]
@anoma I recently installed Nextcloud 10.0.1 in my WEB server (ubuntu 16.04 MySQL 5.7+ & PHP 7.0+ powered), too and as far as i saw till now looks much more stable & user friendly than ownCloud...
After reading those articles http://thenewstack.io/story-behind-nextcloud/ and http://www.techrepublic.com/article/owncloud-founder-has-forked-their-product-into-nextcloud/ and knowing  a co-founder and the main dev. team of ownCloud left to create Nextcloud I have no doubt we will have no other choice than taking a decision soon here about which or what integrate in MiaB.

**Comment by just4t** [comment]
BTW see this forum post about [OwnCloud Vulnerability](https://discourse.mailinabox.email/t/owncloud-vulnerability/1562?u=just4t), too

**Comment by JoshData** [comment]
Now that we have ownCloud 9.1 installed (#894), I'm about ready to make a decision here.

I'm leaning strongly toward Radicale, since we have a proof of concept in #837, plus a yet-to-be-determined calendar front-end. (We would drop support for file sharing and we would drop a web front-end for contacts (at least for now).) We would still need to figure out migration and whether we would need to install both for a period of time.

**Comment by yodax** [comment]
I'm of two minds; 

Keeping ownCloud/nextCloud has the pro that it has cloud storage. That saves me a few euro's in hosting. Also it saves us the work of migrating people to a different backend and making sure people move their files. I think the latest PR for 9.1 makes future upgrades easier/safer since it now also makes back-ups and upgrades people on older versions. 
 
I like Radical since it separates storage from UI. Also it makes miab smaller and hopefully easier to maintain. We don't have to deal with ownCloud upgrades. We don't have to decide if ownCloud or nextCloud has the future.

**Comment by gitacticia** [comment]
@yodax We are using miab as a product we offer on servers we sale. YES ! Without the cloud capacity, we have to resale owncloud and move clients on Gapps. Yes ! I'm using miab as on option "on top of owncloud". It's probably rare, but it exists. I apalogize a lot, but I'm technically with Josh. Radicale seems OK for miab in term of initial design.

Lost the cloud (webdav), will you lost users ?

**Comment by just4t** [comment]
Needed to take a decision last week and finally decided to install 'NextCloud 10.0.1 in my WEB server and runs smoothly then, downgraded the MiAB server and locked frontend access to the /cloud of It.
Knowing 'NextCloud' OFFERS 'calendar' and 'contacts' apps. inegration, too I would be happy with a 'clean' MiaB version.

**Comment by Thomvh** [comment]
Radical seems like a good choice to me. I prefer to separate my mail hosting and my cloud hosting. I already host a Nextcloud next to my Miab server. It works perfectly for now!

**Comment by IWriteThings** [comment]
Playing devils advocate here but I love ownCloud. For what it is worth.

**Comment by bronson** [comment]
Is anyone else a little concerned about the lack of action in Radicale's github repo?  Most recent news is from 11 months ago, not many commits in the past few months.

Seems unlikely that it would be feature complete already!

(for the record: I definitely support moving to something more stable than owncloud)

**Comment by ma0ho** [comment]
I would be happy as well if you keep OwnCloud/Nextcloud, because it allows me to have everything I need to have in the cloud (Mails, Calendars, Contacts, Files) in a single place, which is very convenient.

**Comment by MariusBluem** [comment]
Maybe somebody wants to have a look into the Nextcloud 11 beta, we've just released yesterday:
https://nextcloud.com/blog/two-reasons-why-you-should-test-nextcloud-11-beta-this-weekend/

...also in Nextcloud 10 (the fork-brother of ownCloud 9.1) we had a lot of improvements and fixes. You may also want to have a look into the development in the contacts and calendar app which is going on there very fast ... on the other hand: in ownCloud - nearly all calendar/contacts app maintainers have moved over to Nextcloud. It's time now - time to change. Maybe to Nextcloud 11.

**Comment by IWriteThings** [comment]
Do we have any ideas on what will occur during this change over? I have backups but what is the plan for those of us using owncloud right now. Is there going to be a simple conversion process or will we potentially loose all of our contacts and calendars until we translate over by hand?

Just curious what your thoughts and plans are for this process of switching over?

**Comment by bronson** [comment]
Curious, what change over are you referring to?

I would bet money that the migration will be automatic.  But, until there's an actual plan in place, I don't think anyone can say for sure.

**Comment by IWriteThings** [comment]
@bronson That is all I am worried about. Will migration be automatic? I have backups but the less down time the better... Know what I mean?

**Comment by louwers** [comment]
OwnCloud works very well, so I'm in full support of upgrading to NextCloud 11 once it's released! I don't have much experience with it, because I just got my setup running. Hope a a smooth transition will be possible.

**Comment by scottalanmiller** [comment]
I'm all for NextCloud as well.  That's the obvious choice and what I'd be happiest seeing in use.

**Comment by cromulus** [comment]
my $0.02 is that it is better to use small tools that do one thing and only one thing well.
I'd vote for radicale for CAL/CARD DAV + [seafile](https://www.seafile.com/en/home/) for file sharing.

All that said, I have no experience with Own/NextCloud.

**Comment by guyzmo** [comment]
👍 for small things that just works.

**Comment by anoma** [comment]
We are jaded by an extended period of OwnCloud breaking often. That is not my experience of NextCloud and we need to be careful we are not seeing demons where there are none.

At this point I would suggest the right thing to do would be to move to NextCloud and then "start the clock" again to judge it based on its merits and not those of old OwnCloud.

**Comment by bronson** [comment]
NextCloud is gigantic and feature-laden -- the opposite of MiaB's ideals.  Even if it was 100% reliable (and that would be amazing), it still doesn't seem like it's a very good fit.

**Comment by anoma** [comment]
Oh dont get me wrong, I agree with that but I am simply advocating a "devil you know " approach as a positive interim step in this now, multi year debate, that in reality is no further forward than Aug 2015.

**Comment by mitchellurgero** [comment]
Has anyone looked at Pydio as a possible solution to this issue? (https://pydio.com/)

**Comment by mitchellurgero** [comment]
Nextcloud is also a fork of owncloud with a lot of features and bugs fixed. That might also be a possible solution https://nextcloud.com/features/

List of self-hosted owncloud like software:
http://alternativeto.net/software/owncloud/?platform=self-hosted

**Comment by RobWW** [comment]
I wrote quite a while ago that Owncloud added a lot of usability to Mail-in-a-box and I use its features daily.  Reading through this thread, it appears at least half the commenters are in the same situation.  I am concerned about how other commenters want to causally discard functionality that quite actively used.  For almost all software project, a single person uses only a subset of features -- the problem is that everybody uses a different subset.  Success depends on having a rich enough feature set to attract numerous users.
	
Switching away from Owncloud/Nextcloud seems a lot of work for what will ultimately be a loss of function.  Other have mentioned that Nextcloud has been quite stable and has fixed many organizational problems.  To me, at this point it seems most efficient to give it a try.

I would also like to point out that Owncloud/Nextcloud offers areas to grow.  For example Thunderbird has added the ability to link files instead of sending attachments.  Supporting this feature would be consistent with Mail-in-the-box's goal of promoting privacy and decentralization.  The only problem is the required Owncloud interfaces appears to be disabled at present .

Another potential improvement would be if the Nextcloud email client improves, it could substitute for Roundcube. Although I find Roundcube is quite nice, its lack of integration with contacts and calendars is rather clunky, and Roundcube Next development seems to have ran into problems.   One interface that integrates email, contacts, and calendars make obvious sense and would simplify support.

**Comment by bronson** [comment]
Well, OwnCloud contributes significantly to the maintenance burden of MiaB.  From my perspective, it's caused more wasted time itself than all the other components in the MiaB stack put together.

Taking advantage of more features just makes that worse.

Is there any way to reduce OwnCloud's burden on Mail-in-a-Box maintainers?

**Comment by JoshData** [comment]
>  I am concerned about how other commenters want to causally discard functionality that quite actively used.

No one, absolutely no one, is casual about the loss of functionality. This thread is more than a year old and we still haven't figured things out. And I think I've said that we won't switch without an upgrade path.

But asking for functionality lands on deaf ears if you aren't willing to put in the time to maintain it. Feedback on what's useful is ok... but lots of folks have made the point that ownCloud is useful. There really isn't a need anymore to pile that on.

**Comment by mitchellurgero** [comment]
You could instead use: https://plugins.roundcube.net/packages/melanie2/owncloud would be very simple to use, and would integrate owncloud into roundcube - I can maintain this if it is accepted as a viable option to MIAB.

**Comment by bronson** [comment]
It's not hard to install OwnCloud, it's hard to upgrade it.  You often need to hand-tweak a bunch of esoteric stuff.

**Comment by hex-m** [comment]
OwnCloud (NextCloud) is the easiest solution for the end user. As it's widespread most clients have support for it. The big problem is that they don't play nice with distributions. I would argue that the user of MiaB is the server operator, not the user of that server.

But anyway - we need a working alternative if we consider switching so we should try to agree on what to use _if_ we get rid of OwnCloud.

- [Radicale](https://github.com/Kozea/Radicale) seems like the best bet as a CalDAV/CardDAV Server.
- Roundcube would need a plugin to use the CardDAV backend.
- The most promising Web-Calender seems to be [AgenDAV](https://github.com/adobo/agendav/) but that's not even in the Ubuntu repo yet.
- For file-sync the I see multiple options (that do at least have free software clients for android)
    - git (possibly GitLab) on the server, [Sparkleshare](https://github.com/hbons/SparkleShare) on the client.
    - [Syncthing](https://github.com/syncthing/syncthing) which is in Yakkety.
    - [Seafile](https://github.com/syncthing/syncthing) which is not in Ubuntu yet.

There are some big tasks ahead before we can even think about "replacing ownCloud with something more stable".

**Comment by guyzmo** [comment]
* git+sparkleshare is a neat tool, but it's a bit too complex for the use case, and TBH I don't want to see gitlab shipped with MIAB, it would be totally out of scope for a tool aimed at managing e-mails — and also having a third stack installed for the web, gitlab being based on RoR.
* Syncthing is more intended as a tool to keep filesystem subtrees in sync, not merely as a content repository. What it means is that you won't find easy ways to upload your stuff from your phone or your tablet to your server. But it's great to keep two replicas of your home directory's content on two machines (I use it to keep my documents and development in sync between my laptop and my workstation).
* Seafile really is a dropbox alternative, the same way own/next cloud intend to be.

--------

To give my $.02 on the topic:

Generally speaking, MIAB is about managing email.
* It comes with DNS management, because in 2016 it's a need to properly setup emails.
* Then it does some basic webhosting because it can, and a webservice is needed for webmail and the MIAB configurator. 
* Then CardDav, CalDav are logical addons to e-mails, because those are intemately linked in most Mail User Agents. And Owncloud for a long time has been the best choice to manage that.

So based on that rationale, I see no reason why the file storage feature of owncloud should stay a mandatory feature part of MIAB. I don't think that should come into consideration in the discussion. If you want file syncing? Then spawn a new VM or a new container, and install your tool. Maybe take your chances at installing it alongside MIAB.

**Comment by DeviantEng** [comment]
Personally I have no use for the file storage, though I think an ideal solution would be if a file storage solution could be used for email attachments and somehow automate it.  

Frankly, I've been following this discussion since the start and last commented over a year ago.  A lot of the same ideas/thoughts keep getting repeated, and rarely do I see new information added to the conversation.  I think at this point the dev's need to make a decision on what the plan is.  

But to throw in my current thoughts, I think migrating to NextCloud is the best short-term plan because that would likely be the easiest from a dev-perspective.  Run with NextCloud for a bit and re-evaluate the need to change to something else.

**Comment by JJJ** [comment]
> I think migrating to NextCloud is the best short-term plan because that would likely be the easiest from a dev-perspective. Run with NextCloud for a bit and re-evaluate the need to change to something else.

This is my gut feeling as well, but that's without fully comprehending the migration path from ownCloud to NextCloud.

**Comment by bronson** [comment]
The migration path from OwnCloud to NextCloud is supposed to be easy...  But the OwnCloud migrations were supposed to be easy too.  :-/

**Comment by louwers** [comment]
I'd like to point out you can try out NextCloud yourself at this website: https://demo.nextcloud.com/

I read [here](https://nextcloud.com/migration/) that migration to NextCloud is basically the same as an upgrade of NextCloud... Instructions here: https://docs.nextcloud.com/server/11/admin_manual/maintenance/manual_upgrade.html Any updates on this?

**Comment by JoshData** [comment]
If it's so simple, please try it out with Mail-in-a-Box and report back!

**Comment by MariusBluem** [comment]
I've tested it with Mail-in-a-Box and it is working. 👊 

Have a look at https://nextcloud.com/secure/ ... we have improved the security of the code a lot in the last weeks and months ;) ... At least the complete maintainers team of calendar (@georgehrke and @raghunayyar) has moved over to Nextcloud ✊ ... this said, I don't want to start talking about Nextcloud server's core team 😅 

If you want to learn more about Nextclouds development activity, visit https://nextcloud.com/blog/nextcloud-the-most-active-open-source-file-sync-and-share-project/, and you will see ... why this is the right moment to migrate and move forward 😉 

Nextcloud 11 was released some weeks ago, and is including many improvements on security and stability-side. That said, also the upgrade-progress is now more stable and we experience barely reports about broken instances after upgrades 💪 

You have a working script, and the change would be really as easy as changing some URLs 😜 - Upgrade experience is well. But lets hear some more reports from **your** community. 

With Nextcloud inside of MiaB, you should also consider about shipping the latest `stable11` instead of an hard-coded version ... this prevents from non-updating to maintenance releases 🤔 (MiaB: 9.1.2 - OC: 9.1.3) - just for security reasons. Upgrades of major-versions will be done manually in the code with replacing to `stable12` then. You can do this - whenever you want.

_Disclaimer: I am part of the Nextcloud community. MiaB is running my Mailserver so I know a bit what I am talking about and I am not only a Nextclouder who wants to merge anything for portfolio reasons!_

**Comment by just4t** [comment]
Again, and again, and again. Hope someone will take the final decision about what's next, _before the 2nd birthday of this topic_ ... What I would like to see:

_Option one:_
>> Keep MiaB simple... just DNS's Mger. + Mail Server + Webmail giving the freedom to the end user to use ownCloud, Nextcloud or whatever he would like on a separate web server.

_Option two (perhaps, better for those ones that would like to keep ownCloud) :_
>> Make MiaB install script to ask about what the end user would like to install or better said,  to exclude from installation, aka ownCloud.

Thanks for reading!

**Comment by sudo-gedit** [comment]
I am for option two.  👍 

I would also want a script for the uninstallation of owncloud.

**Comment by anoma** [comment]
If a channel could be opened between between MIAB devs and NextCloud devs whereby both communitys could help fix bugs then I suspect that would change a lot.

Currently at the heart of this is the `MIAB upgrade broke or needs reworked... what caused it... oh its OwnCloud yet again`.

We seem to have 1-2 NextCloud community people here and I think I am right in saying, no OwnCloud people ever?

There is no perfect solution coming any time this decade, we just need to pick an imperfect one so we can move on.

**Comment by calismoonelf** [comment]
> I've tested it with Mail-in-a-Box and it is working. 👊

Did you do anything special? I haven't even begun to look at what is involved, but I have an install that I'm willing to tinker with. Toying with ideas for my company also. I'm getting ready to set up MiaB here for it, would be interested in using short-term secure document sharing.

**Comment by kenthinson** [comment]
Want to add +1 . for switching to next cloud. I only had issues with owncloud it never worked right. I'm now on next cloud 10. I have had 0 issues.

**Comment by xetorixik** [comment]
Any status update on issue 514? 
it's now about 1.5 years ago (at the start) and there is still no decision.

+1 for Radicale  plus a yet-to-be-determined calendar front-end if you ask me.
Nextcloud or Owncloud is a bit overkill for a mail server just like Mailinabox.
But that is obviously personal.

The key seems to me to make a decision

**Comment by kenthinson** [comment]
Just a heads up for anyone interested. I'm now running next cloud 11. I'm using it with roundcube without issues.

**Comment by JoshData** [comment]
Migration is definitely one of the big issues. I haven't been impressed with the security and maintainability of some of the other options though, also. For instance, Radicale's last release posted on its website is from 2015 when a zillion security issues were discovered.

**Comment by xetorixik** [comment]
About the security of- and maintaining Owncloud.
Mail in a box is using 9.1.1, since then 3 new versions are released with load of fixes. (9.1.2, 9.1.3 and 9.1.4).
If security and maintainability is important.
What about Owncloud then?

**Comment by jerogee** [comment]
@JoshData : see @just4t and @kenthinson:  can we PLEASE have an installation prompt that asks whether calendars, filesharing, and a HUGE attack vector called Owncloud/Nextcloud should be installed? The M in MIAB is what it is about. Then I am not talking about how Owncloud/Nextcloud repeatedly causes trouble in MIAB migration/updates.

**Comment by JoshData** [comment]
No prompts. Security is not a choice we're going to give users the ability to opt-out of.

**Comment by jerogee** [comment]
Josh, I perfectly understand where you are coming from, and I understand that a such options would  break the MIAB philosophy. So no problem: I will just keep modding setup/start.sh manually then. (but from a security point of view it is really not a yes/no matter: With each additional service on top of mail-functionality new attach vectors are introduced, and it must be said: Owncloud has a terrible track record in that respect, and I really hope that Nextcloud will not get as notorious)

**Comment by msgerbs** [comment]
Is this confirmation then that you plan to remove owncloud, calendars and everything else he mentioned?On Mar 25, 2017 4:01 PM, Joshua Tauberer <notifications@github.com> wrote:No prompts. Security is not a choice we're going to give users the ability to opt-out of.

**Comment by JoshData** [comment]
No (but I am open to that possibility).

**Comment by yeah** [comment]
@MariusBluem you said you're using Nextcloud on MIAB. Are you using 11+? If so, did you upgrade PHP to 5.6?

**Comment by MariusBluem** [comment]
@yeah I had MiaB with Nextcloud 11 and - you are right - a custom PHP version. In this case, it was 7.0 😉 I had customzied the script a bit - no really big changes 😅 But I think this will not work for MiaB, becasue the maintainer wants to use the PHP version shipped with Ubuntu 14.04.

**Comment by ma0ho** [comment]
Just to bring in a (hopefully) fresh idea to solve that long lasting discussion: My feeling is that we basically have two groups. On one hand we have people that rely on having Owncloud/Nextcloud because they use MIAB as a complete replacement for Googles/Apples/Microsofts services, though mostly for personal use. On the other hand we have people that use it as a mail only solution, which is many cases more a business solution. These people absolutely put security and stability over features and thus do not want Owncloud. I think we will never find a good solution for both groups. Therefore I suggest to split MIAB into two product lines. One, probably maintained by Josh, that focusses on the core without Owncloud. And one that uses that core and puts Owncloud/Nextcloud on top. We would of course need to find a maintainer for that but I'm  certain we will. How about that solution?

**Comment by yeah** [comment]
> Therefore I suggest to split MIAB into two product lines.

I have seen many open source projects fork and it always resulted in alienated/confused users, duplicated work and a general overall loss for the community. I would be strongly against a fork.

**Comment by yeah** [comment]
FWIW I have just implemented Nextcloud support and opened PR #1121.

**Comment by ma0ho** [comment]
@yeah: Thank you for your opinion  on that. Let me explain a bit more 
in detail: I do not regard this as a real fork (although it technically 
is). From my opinion, real forks of projects result from users being 
dissatisfied and trying to do better. This actually results in two 
projects with more or less the same goals (which of course confuses 
users). See Owncloud/Nextcloud as an example.

In this case I think about ONE product that appears in two different 
variants (just as if we had an option in the installer). This could be 
announced in exactly that way, like: Users that need Owncloud/Nextcloud 
install variant A, users that dont't need it, install variant B. If we 
announce that on the same website, I'm certain it will be quite 
transparent and dissatisfy less users that simply removing 
Owncloud/Nextcloud. And we could even go further and leave it as an 
installer question that results in cloning one of the two repositories..

But this is just an idea and I welcome any comments on that.

**Comment by bronson** [comment]
Well, it's been over a year (for me)...  I still don't want owncloud/nextcloud's overfeatured security adventure anywhere near my mailsever.

But Radicale appears to be stalling.  Commits are still landing, but there's no news and the most recent release is from 2015.  No idea what their plan is.

Davical is the other alternative, and it's also not real lively.

Are there any other implementations worth considering?

If not, then I guess Nextcloud is the only way out of this hole...  alas.

**Comment by anoma** [comment]
FWIW it have recently hit my first NextCloud "upgrade broke everything" problem (albeit with the contacts addon). 

Disappointed, I thought those days were gone when I dropped OwnCloud.

But in terms of this project at least we are making progress now thanks to @yeah. So huge thanks for that.

**Comment by yodax** [comment]
@anoma do you have a link to that issue? Would it also have affect MIAB? What was the cause?

**Comment by anoma** [comment]
@yodax https://github.com/nextcloud/contacts/issues/145#issuecomment-287712624

It looks like it may be an edge case but it is just my luck. I use NC locally just for contacts and calendar and nothing more. In the OC days I lived in fear of updating as it broke 2 out of 3 times but this is the first since migrating to NC (in context though I have stock NC + 2 addon total).

I have fixed it but in a really ugly way.

**Comment by zoof** [comment]
@bronson It is not obvious to me why lack of development is a bad thing. I have been using davical for many, many years with no problems.

**Comment by msgerbs** [comment]
@yeah Thank you for taking the initiative to give us at least a temporary resolution to this problem.

**Comment by a-schaefers** [comment]
It should be quite clear that cloud with email is a bad idea. VPS is a lot cheaper now days. Cloud in a box and Mail in a Box should be 2 separate VPS for security reasons. Don't fork the project, instead, fork the installer-- give people a choice which modules or "addons" they want in addition to the MAIL in a box. Offer only support for Mail, make the extra addons unsupported. I came to MIAB because it is easy and takes email security seriously. MIAB needs to get away from cloud and spend the time instead developing docker solutions and perfecting backup solutions and mail archiving.

**Comment by JoshData** [comment]
>  MIAB needs to get away from cloud and spend the time instead developing docker solutions and perfecting backup solutions and mail archiving.

@a-schaefers Great ideas, which of those are you volunteering to help with?

**Comment by a-schaefers** [comment]
@JoshData Don't get me wrong. They're obviously not new ideas and not my ideas. Just wanted to voice my opinion. I instantly love MIAB the moment I installed it. Thank you for all you have done here. I want to help as much as I am able in any way that I can.

**Comment by msgerbs** [comment]
You don't want cloud, calendar, or contacts... But you want them to add docker? What on Earth for?

**Comment by JoshData** [comment]
I'm closing this issue because with the switch to Nextcloud a while back things have gotten better, and I don't foresee having time to find a way to switch to anything else in the near future.
